### ISSUE TITLE
"Variable Name" rules don't handle multibyte characters

### ISSUE THREAD

**Issue by Sega-Zero** [realm/SwiftLint#279]
Having a warning:
> warning: Variable Name Min Length Rule Violation: Variable name should be 3 characters or more: currently 2 characters (variable_name_min_length)

The warning appears on several lines of code in my swift file, but I was able to reproduce this warning only on one code sample in a separate swift file:
 
```swift
protocol SampleProtocol {
    func longMethodNameWithCompletion(completion: () -> Void)
    func longMethodNameWithCompletionAndId(id: Int, completion: () -> Void)
    func longMethodNameWithParameters(param1: Int, param2: Int, strParam: String, completion: () -> Void)
    func yetAnotherMethodNameWithSomeParameters(fromTime: NSDate, toTime: NSDate, accountKey: AccountKey, completion: () -> Void)
}
```
The warning is on the `longMethodNameWithParameters` line.

`.swiftlint.yml`:
```yml
disabled_rules: 
  - line_length
  - trailing_newline
``` 

The file  contains only structs and protocols with markdown comments. Removing `line_length` form `.swiftlint.yml`doesn't help.
Any thoughts?

**Comment by jpsim** [comment]
The `variable_name_min_length` rule considers function parameters as variables, and you have one named `id`, which violates the rule. So this is behaving as expected.

**Comment by Sega-Zero** [comment]
Then why the warning points to the next function?

**Comment by norio-nomura** [comment]
Potentially there is multi byte character issue in most of rules.

**Comment by jpsim** [comment]
When I copy the text from your sample above, and run SwiftLint on it, the violation occurs on line 3, character 44, which is correct:

```
$ pbpaste | swiftlint lint --use-stdin
<nopath>:6: warning: Trailing Newline Violation: Files should have a single trailing newline. (trailing_newline)
<nopath>:4: warning: Line Length Violation: Line should be 100 characters or less: currently 105 characters (line_length)
<nopath>:5: warning: Line Length Violation: Line should be 100 characters or less: currently 129 characters (line_length)
<nopath>:3:44: warning: Variable Name Min Length Rule Violation: Variable name should be 3 characters or more: currently 2 characters (variable_name_min_length)
Done linting! Found 4 violations, 0 serious in 1 file.
```

So I suspect that @norio-nomura is right, as always :wink:, that there are some multibyte characters in your file that you have not shared with us. Could you please share a complete reproducible sample with us so we can repro and fix this? Thanks!

**Comment by Sega-Zero** [comment]
```swift
//
//  Experiments.swift
//  Project
//
//  Created by Сергей Галездинов on 17.12.15.
//  Copyright © 2015 Сергей Галездинов. All rights reserved.
//

import Foundation
import UIKit

protocol SampleProtocol {
    func longMethodNameWithCompletion(completion: () -> Void)
    func longMethodNameWithCompletionAndId(id: Int, completion: () -> Void)
    func longMethodNameWithParameters(param1: Int, param2: Int, strParam: String, completion: () -> Void)
    func yetAnotherMethodNameWithSomeParameters(fromTime: NSDate, toTime: NSDate, accountKey: AccountKey, completion: () -> Void)
}
```

The file contains russian symbols

**Comment by jpsim** [comment]
From what I can tell, it seems like `Location.init(file:offset:)` uses `NSString.lineAndCharacterForCharacterOffset(_:)` although the offset it's getting for the variable name family of rules is a byte offset, not a character offset.
